### PR TITLE
Replace gogs/cron with go-co-op/gocron

### PR DESCRIPTION
**Pull request by harryzcy**
Replace `github.com/gogs/cron` with `github.com/go-co-op/gocron` as the former package is not maintained for many years.

### CODE REVIEW

File: services/cron/cron.go [hunk 1]
```diff
@@ -77,24 +77,30 @@ type TaskTable []*TaskTableRow
...
+	for _, job := range jobs {
+		// the first tag is the task name
+		jobMap[job.Tags()[0]] = job
 	}
+
 	lock.Lock()
 	defer lock.Unlock()
+	for _, job := range jobs {
+		job.PreviousRun()
+	}
```
Comment (delvh) [inline services/cron/cron.go]: ?

File: services/cron/cron.go [hunk 2]
```diff
@@ -14,10 +14,10 @@ import (
 	"code.gitea.io/gitea/modules/sync"
 	"code.gitea.io/gitea/modules/translation"
 
-	"github.com/gogs/cron"
+	"github.com/go-co-op/gocron"
 )
 
-var c = cron.New()
+var s = gocron.NewScheduler(time.Local)
```
Comment (silverwind) [inline services/cron/cron.go]: Name this variable `scheduler`, more readable.

**Review by silverwind — APPROVED** [review]

**Comment by harryzcy** [comment]
The tests seems related

**Comment by harryzcy** [comment]
The failing test is resolved now

**Review by lunny — APPROVED** [review]

File: services/cron/tasks.go [hunk 3]
```diff
@@ -199,3 +198,33 @@ func RegisterTaskFatal(name string, config Config, fun func(context.Context, *us
...
+		if c == ' ' {
+			if !previousIsSpace {
+				numSpaceBlocks++
+				previousIsSpace = true
+			}
+		} else {
+			previousIsSpace = false
+		}
+	}
+	return numSpaceBlocks == 5
```
Comment (delvh) [inline services/cron/tasks.go]:
```suggestion
	return len(strings.Fields(schedule)) >= 6
```

**Pull request merged**